cmd/kubeadm/app/util/output/output.go

```diff
@@ -0,0 +1,216 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package output
+
+import (
+	"fmt"
+	"io"
+	"strings"
+
+	"github.com/spf13/cobra"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/cli-runtime/pkg/genericclioptions"
+	"k8s.io/cli-runtime/pkg/printers"
+	"k8s.io/klog"
+)
+
+// KubeadmTextPrintFlags is an iterface to handle custom text output
+type KubeadmTextPrintFlags interface {
+	ToPrinter(outputFormat string) (KubeadmPrinter, error)
+}
+
+// KubeadmOutputFlags composes common printer flag structs
+// used across kubeadm commands, and provides a method
+// of retrieving a known printer based on flag values provided.
+type KubeadmOutputFlags struct {
```

Comment:
*Flags suffix is used in the [cli-runtime code](https://github.com/kubernetes/cli-runtime/blob/master/pkg/genericclioptions/json_yaml_flags.go#L37). I'd rather use their naming convention.

P.S.: Would s/OutputFlags/OutputPrintFlags/ go? cli runtime uses  PrintFlags suffix, so this way it would be even closer to their names.

Reply:
but the structure seems to contain more than just flags?

Reply:
Yes, it contains flags and additional fields to serve its second purpose that's mentioned in the comment - to retrive a printer based on the flag values.